sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java

```diff
@@ -537,60 +540,143 @@ public PollerFlux<ShareFileCopyInfo, Void> beginCopy(String sourceUrl, FileSmbPr
-        String fileCreationTime = FileSmbProperties.parseFileSMBDate(tempSmbProperties.getFileCreationTime());
-        String fileLastWriteTime = FileSmbProperties.parseFileSMBDate(tempSmbProperties.getFileLastWriteTime());
-
-        if (filePermissionCopyMode == null || filePermissionCopyMode == PermissionCopyModeType.SOURCE) {
-            if (filePermission != null || filePermissionKey != null) {
+        if (options.getFilePermission() == null || options.getPermissionCopyModeType() == PermissionCopyModeType.SOURCE) {
+            if ((options.getFilePermission() != null || filePermissionKey != null) && options.getPermissionCopyModeType() != PermissionCopyModeType.OVERRIDE) {
                 return PollerFlux.error(LOGGER.logExceptionAsError(new IllegalArgumentException(
                     "File permission and file permission key can not be set when PermissionCopyModeType is source or "
                         + "null")));
             }
-        } else if (filePermissionCopyMode == PermissionCopyModeType.OVERRIDE) {
+        } else if (options.getPermissionCopyModeType() == PermissionCopyModeType.OVERRIDE) {
             // Checks that file permission and file permission key are valid
             try {
-                validateFilePermissionAndKey(filePermission, tempSmbProperties.getFilePermissionKey());
+                validateFilePermissionAndKey(options.getFilePermission(), tempSmbProperties.getFilePermissionKey());
             } catch (RuntimeException ex) {
                 return PollerFlux.error(LOGGER.logExceptionAsError(ex));
             }
         }
 
+        // check if only copy flag or smb properties are set (not both)
+        CopyableFileSmbPropertiesList list = options.getSmbPropertiesToCopy()  == null ? new CopyableFileSmbPropertiesList() : options.getSmbPropertiesToCopy();
+        if (list.isFileAttributes() && tempSmbProperties.getNtfsFileAttributes() != null) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException("Both CopyableFileSmbPropertiesList.isSetFileAttributes and smbProperties.ntfsFileAttributes cannot be set."));
+        }
+        if (list.isCreatedOn() && tempSmbProperties.getFileCreationTime() != null) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException("Both CopyableFileSmbPropertiesList.isSetCreatedOn and smbProperties.fileCreationTime cannot be set."));
+        }
+        if (list.isLastWrittenOn() && tempSmbProperties.getFileLastWriteTime() != null) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException("Both CopyableFileSmbPropertiesList.isSetLastWrittenOn and smbProperties.fileLastWriteTime cannot be set."));
+        }
+        if (list.isChangedOn() && tempSmbProperties.getFileChangeTime() != null) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException("Both CopyableFileSmbPropertiesList.isSetChangedOn and smbProperties.fileChangeTime cannot be set."));
+        }
```

Comment:
could we rely on service for this validation ?
otherwise could we use `CopyableFileSmbPropertiesList` ctors to allow only valid combinations?

Reply:
We also have this check for .NET - https://github.com/Azure/azure-sdk-for-net/blob/ff1ef6a4bbc0286fbaebbc458ce569ce85f7d9b1/sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs#L1419.

I believe the service doesn't return a friendly error message in this case.